install.sh

```
@@ -1475,8 +1578,32 @@ check_prerequisites() {
+
+        # Build detailed diagnostic information about port conflicts
+        local ports_json="["
+        for i in "${!failed_ports[@]}"; do
+            if [ $i -gt 0 ]; then
+                ports_json+=","
+            fi
+            ports_json+="{\"port\":${failed_ports[$i]},\"process\":\"$(echo "${port_processes[$i]}" | sed 's/"/\\"/g')\"}"
+        done
+        ports_json+="]"
+
+        local diagnostic_json=$(cat <<EOF
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)